core/src/databases/database.rs

```diff
@@ -83,16 +85,216 @@ impl Database {
+        project: &Project,
+        store: Box<dyn Store + Sync + Send>,
+    ) -> Result<Connection> {
+        match self.db_type {
+            DatabaseType::REMOTE => Err(anyhow!(
+                "Cannot build an in-memory SQLite DB for a remote database."
+            )),
+            DatabaseType::LOCAL => {
+                let time_build_db_start = utils::now();
+                let (schema, rows_by_table) = self.get_schema(project, store.clone(), true).await?;
+                let rows_by_table = match rows_by_table {
+                    Some(rows) => rows,
+                    None => return Err(anyhow!("No rows found")),
+                };
+                utils::done(&format!(
+                    "DSSTRUCTSTAT Finished retrieving schema: duration={}ms",
+                    utils::now() - time_build_db_start
+                ));
+
+                let table_schemas: HashMap<String, TableSchema> = schema
+                    .iter()
+                    .filter(|(_, table)| !table.schema.is_empty())
+                    .map(|(table_name, table)| (table_name.clone(), table.schema.clone()))
+                    .collect();
+
+                let generate_create_table_sql_start = utils::now();
+                let create_tables_sql: String = schema
+                    .iter()
+                    .filter(|(_, table)| !table.schema.is_empty())
+                    .map(|(table_name, table)| {
+                        table
+                            .schema
+                            .get_create_table_sql_string(table_name.as_str())
+                    })
+                    .collect::<Vec<_>>()
+                    .join("\n");
+                utils::done(&format!(
+                    "DSSTRUCTSTAT Finished generating create table SQL: duration={}ms",
+                    utils::now() - generate_create_table_sql_start
+                ));
+
+                let conn = rusqlite::Connection::open_in_memory()?;
+
+                let create_tables_execute_start = utils::now();
+                conn.execute_batch(&create_tables_sql)?;
+                utils::done(&format!(
+                    "DSSTRUCTSTAT Finished creating tables: duration={}ms",
+                    utils::now() - create_tables_execute_start
+                ));
+
+                let insert_execute_start = utils::now();
+                rows_by_table
+                    .iter()
+                    .filter(|(_, rows)| !rows.is_empty())
+                    .map(|(table_name, rows)| {
+                        let table_schema = match table_schemas.get(table_name) {
+                            Some(s) => Ok(s),
+                            None => Err(anyhow!("No schema found for table {}", table_name)),
+                        }?;
+
+                        let (sql, field_names) = table_schema.get_insert_sql(table_name);
+                        let mut stmt = conn.prepare(&sql)?;
+
+                        let params: Vec<Vec<Value>> = rows
```

Comment:
this let params and then params.iter() can be removed.

Reply:
It was needed for a lifetime issue IIRC. Can look into it.

Reply:
Ok, managed to remove. However, I still need to `collect` + `iter` to get out of the `par_iter`.